plenario/api/condition_builder.py

```diff
@@ -20,80 +21,78 @@
+    :param ctree: dictionary of conditions created from JSON
 
-        if op in field_ops:
-            return _operator_to_condition(column, op, value)
+    :returns SQLAlchemy conditions for querying the table with"""
 
+    op = ctree['op']
 
-def _operator_to_condition(column, operator, operand):
+    if op == "and":
+        return and_(
+            _parse_condition_tree(table, child, literally)
+            for child in ctree['val']
+        )
+
+    elif op == "or":
+        return or_(
+            _parse_condition_tree(table, child, literally)
+            for child in ctree['val']
+        )
+
+    elif op in field_ops:
+        col = ctree['col']
+        val = ctree['val']
+        try:
+            return _operator_to_condition(
+                getattr(table, 'columns')[col], op, val, literally
+            )
+
+        # Exists for date__time_of_day. Since it doesn't come
+        except KeyError:
+            return getattr(col, field_ops[op])(val)
+
+
+def _operator_to_condition(column, operator, operand, literally=False):
     """Convert an operation into a SQLAlchemy condition. Operators
     are mapped to SQLAlchemy methods with the field_ops dictionary.
 
     :param column: column object from the table to build condition for
     :param operator: string name of the desired operator
     :param operand: some target value or parameter
+    :param literally: return condition as a string literal
 
-    :returns a string SQLAlchemy condition"""
+    :returns: SQLAlchemy condition or string"""
 
     if operator == 'in':
-        return column.in_(operand.split(','))
+        condition = column.in_(operand.split(','))
     elif operator == 'eq':
-        return column == operand
+        condition = column == operand
     else:
-        return getattr(column, field_ops[operator])(operand)
-
-
-# general_filters
-# ===============
-# Filters which apply to all the tables, but whose columns are not directly
-# specified by an argument key. ex: 'obs_date__ge' translates into a condition
-# that needs to be made for a table's 'point_date' column.
-
-general_filters = {
-    'obs_date__ge':
-        lambda table, value:  table.c.point_date >= value,
-    'obs_date__le':
-        lambda table, value:  table.c.point_date <= value,
-    'date__time_of_day_ge':
-        lambda table, value:  date__time_of_day_filter(table, 'ge', value),
-    'date__time_of_day_le':
-        lambda table, value:  date__time_of_day_filter(table, 'le', value),
-    'geom':
-        lambda table, value:  table.c.geom.ST_Within(func.ST_GeomFromGeoJSON(value))
-}
-
-
-def date__time_of_day_filter(table, op, val):
-    """Because I couldn't fit it into a one line lambda.
-
-    :param table: SQLAlchemy table object
-    :param op: string op code
-    :param val: target value to be compared against
+        condition = getattr(column, field_ops[operator])(operand)
 
-    :returns: table condition for just the hour"""
+    if literally:
```

Comment:
This exists specifically for the `/shapes` endpoint, specifically the `export_shape` method. This method relies on a subprocess call made to ogr2ogr, which uses a literal string SQL query to fetch data from postgres. These lines here enable `_operator_to_condition` to return the condition together as a string with its target values, instead of a SQLAlchemy condition with placeholders